g++ 4.2.1 on Mac OS X 10.5 doesn't seem to honor the following line:

#   pragma GCC diagnostic warning "-Wunused-function"

So the following error results:

/usr/bin/g++-4.2 -c -I. -I../../include -I../../../include  -g -O2 -DNDEBUG -Wall -Werror -D_REENTRANT Scanner.cpp
cc1plus: warnings being treated as errors
lex.yy.c:1129: warning: ?void yyunput(int, char*)? defined but not used
make[3]: *** [Scanner.o] Error 1

The patch only comments out the declaration and definition of yyunput,
it doesn't comment out

#define unput(c) yyunput( c, (yytext_ptr)  )

because if it is unused, then unput() should not be inadvertently
used.


diff -ru Ice-3.5.0.orig/cpp/test/Freeze/complex/Scanner.cpp Ice-3.5.0/cpp/test/Freeze/complex/Scanner.cpp
--- Ice-3.5.0.orig/cpp/test/Freeze/complex/Scanner.cpp	2013-03-11 08:19:46.000000000 -0700
+++ Ice-3.5.0/cpp/test/Freeze/complex/Scanner.cpp	2013-03-14 19:43:54.497251598 -0700
@@ -569,7 +569,9 @@
 #endif
 #endif
 
+#if 0
     static void yyunput (int c,char *buf_ptr  );
+#endif
     
 #ifndef yytext_ptr
 static void yy_flex_strncpy (char *,yyconst char *,int );
@@ -1127,6 +1129,7 @@
 	return yy_is_jam ? 0 : yy_current_state;
 }
 
+#if 0
     static void yyunput (int c, register char * yy_bp )
 {
 	register char *yy_cp;
@@ -1163,6 +1166,7 @@
 	(yy_hold_char) = *yy_cp;
 	(yy_c_buf_p) = yy_cp;
 }
+#endif
 
 #ifndef YY_NO_INPUT
 #ifdef __cplusplus
